From 224b28d337f36a937b34685519d7b657974a95d0 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Tue, 3 Aug 2021 10:46:56 +0200
Subject: [PATCH 04/39] Remove RHEL 7.1.0 machine type

RH-Author: quintela1 <quintela@redhat.com>
RH-MergeRequest: 30: Remove RHEL 7.x.0 machine types
RH-Commit: [2/6] af1d8896d63756d1acc5af8658465808c4a13ca3 (juan.quintela/qemu-kvm)
RH-Bugzilla: 1968519
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

migrate_pre_2_2 was a redhat artifact, removed.
broken_reserved_end is already fixed, nothing to do.

Signed-off-by: Juan Quintela <quintela@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/char/serial.c      | 12 ----------
 hw/core/machine.c     | 23 -------------------
 hw/i386/pc.c          | 51 -------------------------------------------
 hw/i386/pc_piix.c     | 41 ----------------------------------
 hw/rtc/mc146818rtc.c  |  4 ----
 include/hw/boards.h   |  3 ---
 include/hw/i386/pc.h  |  3 ---
 migration/migration.c |  2 --
 migration/migration.h |  5 -----
 9 files changed, 144 deletions(-)

diff --git a/hw/char/serial.c b/hw/char/serial.c
index cc378142a3..3e4344cb2a 100644
--- a/hw/char/serial.c
+++ b/hw/char/serial.c
@@ -690,9 +690,6 @@ static int serial_post_load(void *opaque, int version_id)
 static bool serial_thr_ipending_needed(void *opaque)
 {
     SerialState *s = opaque;
-    if (migrate_pre_2_2) {
-        return false;
-    }
 
     if (s->ier & UART_IER_THRI) {
         bool expected_value = ((s->iir & UART_IIR_ID) == UART_IIR_THRI);
@@ -774,9 +771,6 @@ static const VMStateDescription vmstate_serial_xmit_fifo = {
 static bool serial_fifo_timeout_timer_needed(void *opaque)
 {
     SerialState *s = (SerialState *)opaque;
-    if (migrate_pre_2_2) {
-        return false;
-    }
 
     return timer_pending(s->fifo_timeout_timer);
 }
@@ -795,9 +789,6 @@ static const VMStateDescription vmstate_serial_fifo_timeout_timer = {
 static bool serial_timeout_ipending_needed(void *opaque)
 {
     SerialState *s = (SerialState *)opaque;
-    if (migrate_pre_2_2) {
-        return false;
-    }
 
     return s->timeout_ipending != 0;
 }
@@ -816,9 +807,6 @@ static const VMStateDescription vmstate_serial_timeout_ipending = {
 static bool serial_poll_needed(void *opaque)
 {
     SerialState *s = (SerialState *)opaque;
-    if (migrate_pre_2_2) {
-        return false;
-    }
 
     return s->poll_msl >= 0;
 }
diff --git a/hw/core/machine.c b/hw/core/machine.c
index 6c534e14fa..cc7c826593 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -270,29 +270,6 @@ GlobalProperty hw_compat_rhel_7_2[] = {
 };
 const size_t hw_compat_rhel_7_2_len = G_N_ELEMENTS(hw_compat_rhel_7_2);
 
-/* Mostly like hw_compat_2_1 but:
- *    we don't need virtio-scsi-pci since 7.0 already had that on
- *
- * RH: Note, qemu-extended-regs should have been enabled in the 7.1
- * machine type, but was accidentally turned off in 7.2 onwards.
- */
-GlobalProperty hw_compat_rhel_7_1[] = {
-    { "intel-hda-generic", "old_msi_addr", "on" },
-    { "VGA", "qemu-extended-regs", "off" },
-    { "secondary-vga", "qemu-extended-regs", "off" },
-    { "usb-mouse", "usb_version", stringify(1) },
-    { "usb-kbd", "usb_version", stringify(1) },
-    { "virtio-pci",  "virtio-pci-bus-master-bug-migration", "on" },
-    { "virtio-blk-pci", "any_layout", "off" },
-    { "virtio-balloon-pci", "any_layout", "off" },
-    { "virtio-serial-pci", "any_layout", "off" },
-    { "virtio-9p-pci", "any_layout", "off" },
-    { "virtio-rng-pci", "any_layout", "off" },
-    /* HW_COMPAT_RHEL7_1 - introduced with 2.10.0 */
-    { "migration", "send-configuration", "off" },
-};
-const size_t hw_compat_rhel_7_1_len = G_N_ELEMENTS(hw_compat_rhel_7_1);
-
 GlobalProperty hw_compat_5_2[] = {
     { "ICH9-LPC", "smm-compat", "on"},
     { "PIIX4_PM", "smm-compat", "on"},
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 516ca50353..3c1f5be4fa 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -541,57 +541,6 @@ GlobalProperty pc_rhel_7_2_compat[] = {
 };
 const size_t pc_rhel_7_2_compat_len = G_N_ELEMENTS(pc_rhel_7_2_compat);
 
-GlobalProperty pc_rhel_7_1_compat[] = {
-    { "kvm64" "-" TYPE_X86_CPU, "vme", "off" },
-    { "kvm32" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Conroe" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Penryn" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Nehalem" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Nehalem-IBRS" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Westmere" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Westmere-IBRS" "-" TYPE_X86_CPU, "vme", "off" },
-    { "SandyBridge" "-" TYPE_X86_CPU, "vme", "off" },
-    { "SandyBridge-IBRS" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Haswell" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Haswell-IBRS" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Broadwell" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Broadwell-IBRS" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Opteron_G1" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Opteron_G2" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Opteron_G3" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Opteron_G4" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Opteron_G5" "-" TYPE_X86_CPU, "vme", "off" },
-    { "Haswell" "-" TYPE_X86_CPU,  "f16c", "off" },
-    { "Haswell-IBRS" "-" TYPE_X86_CPU, "f16c", "off" },
-    { "Haswell" "-" TYPE_X86_CPU, "rdrand", "off" },
-    { "Haswell-IBRS" "-" TYPE_X86_CPU, "rdrand", "off" },
-    { "Broadwell" "-" TYPE_X86_CPU, "f16c", "off" },
-    { "Broadwell-IBRS" "-" TYPE_X86_CPU, "f16c", "off" },
-    { "Broadwell" "-" TYPE_X86_CPU, "rdrand", "off" },
-    { "Broadwell-IBRS" "-" TYPE_X86_CPU, "rdrand", "off" },
-    { "coreduo" "-" TYPE_X86_CPU, "vmx", "on" },
-    { "core2duo" "-" TYPE_X86_CPU, "vmx", "on" },
-    { "qemu64" "-" TYPE_X86_CPU, "min-level", stringify(4) },
-    { "kvm64" "-" TYPE_X86_CPU, "min-level", stringify(5) },
-    { "pentium3" "-" TYPE_X86_CPU, "min-level", stringify(2) },
-    { "n270" "-" TYPE_X86_CPU, "min-level", stringify(5) },
-    { "Conroe" "-" TYPE_X86_CPU, "min-level", stringify(4) },
-    { "Penryn" "-" TYPE_X86_CPU, "min-level", stringify(4) },
-    { "Nehalem" "-" TYPE_X86_CPU, "min-level", stringify(4) },
-    { "n270" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Penryn" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Conroe" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Nehalem" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Westmere" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "SandyBridge" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "IvyBridge" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Haswell" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Haswell-noTSX" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Broadwell" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-    { "Broadwell-noTSX" "-" TYPE_X86_CPU, "min-xlevel", stringify(0x8000000a) },
-};
-const size_t pc_rhel_7_1_compat_len = G_N_ELEMENTS(pc_rhel_7_1_compat);
-
 /*
  * The PC_RHEL_*_COMPAT serve the same purpose for RHEL-7 machine
  * types as the PC_COMPAT_* do for upstream types.
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index b4e8034671..2a6a28f1e5 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1075,44 +1075,3 @@ static void pc_machine_rhel720_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(rhel720, "pc-i440fx-rhel7.2.0", pc_init_rhel720,
                   pc_machine_rhel720_options);
-
-static void pc_compat_rhel710(MachineState *machine)
-{
-    PCMachineState *pcms = PC_MACHINE(machine);
-    PCMachineClass *pcmc = PC_MACHINE_GET_CLASS(pcms);
-
-    /* From pc_compat_2_2 */
-    pcmc->rsdp_in_ram = false;
-    machine->suppress_vmdesc = true;
-
-    /* From pc_compat_2_1 */
-    pcmc->smbios_uuid_encoded = false;
-    x86_cpu_change_kvm_default("svm", NULL);
-    pcmc->enforce_aligned_dimm = false;
-
-    /* Disable all the extra subsections that were added in 2.2 */
-    migrate_pre_2_2 = true;
-
-    /* From pc_i440fx_2_4_machine_options */
-    pcmc->broken_reserved_end = true;
-}
-
-static void pc_init_rhel710(MachineState *machine)
-{
-    pc_compat_rhel710(machine);
-    pc_init1(machine, TYPE_I440FX_PCI_HOST_BRIDGE, \
-             TYPE_I440FX_PCI_DEVICE);
-}
-
-static void pc_machine_rhel710_options(MachineClass *m)
-{
-    pc_machine_rhel720_options(m);
-    m->family = "pc_piix_Y";
-    m->desc = "RHEL 7.1.0 PC (i440FX + PIIX, 1996)";
-    m->default_display = "cirrus";
-    compat_props_add(m->compat_props, hw_compat_rhel_7_1, hw_compat_rhel_7_1_len);
-    compat_props_add(m->compat_props, pc_rhel_7_1_compat, pc_rhel_7_1_compat_len);
-}
-
-DEFINE_PC_MACHINE(rhel710, "pc-i440fx-rhel7.1.0", pc_init_rhel710,
-                  pc_machine_rhel710_options);
diff --git a/hw/rtc/mc146818rtc.c b/hw/rtc/mc146818rtc.c
index 4a2e52031b..3a70a988e1 100644
--- a/hw/rtc/mc146818rtc.c
+++ b/hw/rtc/mc146818rtc.c
@@ -824,10 +824,6 @@ static bool rtc_irq_reinject_on_ack_count_needed(void *opaque)
 {
     RTCState *s = (RTCState *)opaque;
 
-    if (migrate_pre_2_2) {
-        return false;
-    }
-
     return s->irq_reinject_on_ack_count != 0;
 }
 
diff --git a/include/hw/boards.h b/include/hw/boards.h
index 90ae100bfc..070db0b9b1 100644
--- a/include/hw/boards.h
+++ b/include/hw/boards.h
@@ -445,7 +445,4 @@ extern const size_t hw_compat_rhel_7_3_len;
 extern GlobalProperty hw_compat_rhel_7_2[];
 extern const size_t hw_compat_rhel_7_2_len;
 
-extern GlobalProperty hw_compat_rhel_7_1[];
-extern const size_t hw_compat_rhel_7_1_len;
-
 #endif
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 2d17daf71f..84d985f57c 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -311,9 +311,6 @@ extern const size_t pc_rhel_7_3_compat_len;
 extern GlobalProperty pc_rhel_7_2_compat[];
 extern const size_t pc_rhel_7_2_compat_len;
 
-extern GlobalProperty pc_rhel_7_1_compat[];
-extern const size_t pc_rhel_7_1_compat_len;
-
 /* Helper for setting model-id for CPU models that changed model-id
  * depending on QEMU versions up to QEMU 2.4.
  */
diff --git a/migration/migration.c b/migration/migration.c
index 9d185f0e28..9cf1cde39d 100644
--- a/migration/migration.c
+++ b/migration/migration.c
@@ -167,8 +167,6 @@ INITIALIZE_MIGRATE_CAPS_SET(check_caps_background_snapshot,
     MIGRATION_CAPABILITY_X_COLO,
     MIGRATION_CAPABILITY_VALIDATE_UUID);
 
-bool migrate_pre_2_2;
-
 /* When we add fault tolerance, we could have several
    migrations at once.  For now we don't need to add
    dynamic creation of migration */
diff --git a/migration/migration.h b/migration/migration.h
index 1b6c69751c..db6708326b 100644
--- a/migration/migration.h
+++ b/migration/migration.h
@@ -368,11 +368,6 @@ bool check_dirty_bitmap_mig_alias_map(const BitmapMigrationNodeAliasList *bbm,
 void migrate_add_address(SocketAddress *address);
 
 int foreach_not_ignored_block(RAMBlockIterFunc func, void *opaque);
-/*
- * Disables a load of subsections that were added in 2.2/rh7.2 for backwards
- * migration compatibility.
- */
-extern bool migrate_pre_2_2;
 
 #define qemu_ram_foreach_block \
   #warning "Use foreach_not_ignored_block in migration code"
-- 
2.27.0

